cfme/fixtures/service_fixtures.py

```diff
@@ -225,3 +229,36 @@ def generic_service(appliance, generic_catalog_item):
         service.delete()
     if provision_request.exists:
         provision_request.remove_request(method="rest")
+
+
+@pytest.fixture()
+def import_dialog(appliance, file_name):
+    """This fixture will help to import dialog file."""
+
+    # Download dialog file from FTP server
+    fs = FTPClientWrapper(cfme_data.ftpserver.entities.datastores)
+    file_path = fs.download(file_name)
+
+    # Import dialog yml to appliance
+    import_export = DialogImportExport(appliance)
+    import_export.import_dialog(file_path)
+
+    # Read yml to get the field name
+    with open(file_path, "r") as stream:
+        dialog = yaml.load(stream, Loader=yaml.BaseLoader)
+        # It returns list of dicts
+        ele_label = dialog[0]['dialog_tabs'][0]['dialog_groups'][0]['dialog_fields'][0]['name']
+
+    # File name contains 'yml' , replacing it
+    sd = appliance.collections.service_dialogs.instantiate(label=file_name.replace(".yml", ""))
+    yield sd, ele_label
+    sd.delete()
+
+
+@pytest.fixture(scope="function")
+def catalog_item_with_imported_dialog(appliance, provider, provisioning, import_dialog, catalog):
+    """Catalog Item with imported dialog"""
+    catalog_item = create_catalog_item(appliance, provider, provisioning, import_dialog[0], catalog)
+    yield catalog_item, import_dialog[1]
+    catalog_item.delete()
```

Comment:
I would check `exists` status before `deletion`. Its global fixture, may be we use it and preform some deletion operation before teardown...